gravitee-apim-console-webui/src/management/settings/portal-ng/portal-ng.component.spec.ts

```diff
@@ -0,0 +1,926 @@
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { HttpTestingController } from '@angular/common/http/testing';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
+import { InteractivityChecker } from '@angular/cdk/a11y';
+import { GioFormTagsInputHarness, GioSaveBarHarness } from '@gravitee/ui-particles-angular';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatSlideToggleHarness } from '@angular/material/slide-toggle/testing';
+
+import { PortalNgComponent } from './portal-ng.component';
+import { PortalNgModule } from './portal-ng.module';
+
+import { CONSTANTS_TESTING, GioTestingModule } from '../../../shared/testing';
+import { GioTestingPermissionProvider } from '../../../shared/components/gio-permission/gio-permission.service';
+import { fakePortalSettings } from '../../../entities/portal/portalSettings.fixture';
+
+describe('PortalNgComponent', () => {
+  let fixture: ComponentFixture<PortalNgComponent>;
+  let loader: HarnessLoader;
+  let httpTestingController: HttpTestingController;
+  const portalSettingsMock = fakePortalSettings();
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, GioTestingModule, PortalNgModule, MatIconTestingModule],
+      providers: [
+        {
+          provide: GioTestingPermissionProvider,
+          useValue: ['environment-settings-u'],
+        },
+      ],
+    }).overrideProvider(InteractivityChecker, {
+      useValue: {
+        isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        isTabbable: () => true, // This traps focus checks and so avoid warnings when dealing with
+      },
+    });
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PortalNgComponent);
+    loader = TestbedHarnessEnvironment.loader(fixture);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    fixture.detectChanges();
+    expectPortalSettingsGetRequest(portalSettingsMock);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    httpTestingController.verify();
+  });
+
+  describe('Portal settings form', () => {
+    it('display settings form and edit Company field', async () => {
+      const saveBar = await loader.getHarness(GioSaveBarHarness);
+      expect(await saveBar.isVisible()).toBe(false);
+
+      const companyInput = await loader.getHarness(MatInputHarness.with({ selector: '[formControlName=apikeyHeader' }));
+      await companyInput.setValue('Test company');
+      expect(await saveBar.isSubmitButtonInvalid()).toEqual(false);
+      await saveBar.clickSubmit();
+
+      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.env.baseURL}/settings`);
+      expect(req.request.method).toEqual('POST');
+      expect(req.request.body).toEqual({
```

Comment:
You can return a `fakePortalSettings` with the api key header attribute modified. Will be easier to read rather than a giant object 👍

Reply:
Done